File: P1A11_Termometro/Python/testTermometro.py
from Termometro import Termometro

def main():
    # 1. Instancia con constructor por defecto (aleatorio)
    print("--- Termómetro 1 (Aleatorio) ---")
    t1 = Termometro()
    t1.mostrarTemperaturas()
    print("-" * 30)

    # 2. Instancia con constructor parametrizado (usuario)
    print("--- Termómetro 2 (Usuario) ---")
    try:
        temp_usuario = float(input("Ingrese la temperatura en Celsius: "))
        t2 = Termometro(temp_usuario)
        t2.mostrarTemperaturas()
        print("-" * 30)
    except ValueError:
        print("Error: Por favor ingrese un número válido.")

if __name__ == "__main__":
    main()

File: P1A11_Termometro/Java/testTermometro.java
package P1A11_Termometro.Java;

import java.util.Scanner;

public class testTermometro {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // 1. Instancia con constructor por defecto (aleatorio)
        System.out.println("--- Termometro 1 (Aleatorio) ---");
        Termometro t1 = new Termometro();
        t1.mostrarTemperaturas();
        System.out.println("------------------------------");

        // 2. Instancia con constructor parametrizado (usuario)
        System.out.println("--- Termometro 2 (Usuario) ---");
        System.out.print("Ingrese la temperatura en Celsius: ");
        
        if (sc.hasNextDouble()) {
            double tempUsuario = sc.nextDouble(); // Leer temperatura del usuario
            Termometro t2 = new Termometro(tempUsuario); // Usar constructor parametrizado
            t2.mostrarTemperaturas();
        } else {
            System.out.println("Error: Por favor ingrese un numero valido.");
        }
        System.out.println("------------------------------");
        
        sc.close();
    }
}